### PR TITLE
[Merged by Bors] - Add an example to draw a rectangle

### PR DESCRIPTION
**Pull request by r00ster91**
# Objective

Every time I come back to Bevy I face the same issue: how do I draw a rectangle again? How did that work? So I go to https://github.com/bevyengine/bevy/tree/main/examples in the hope of finding literally the simplest possible example that draws something on the screen without any dependency such as an image. I don't want to have to add some image first, I just quickly want to get something on the screen with `main.rs` alone so that I can continue building on from that point on. Such an example is particularly helpful for a quick start for smaller projects that don't even need any assets such as images (this is my case currently).

Currently every single example of https://github.com/bevyengine/bevy/tree/main/examples#2d-rendering (which is the first section after hello world that beginners will look for for very minimalistic and quick examples) depends on at least an asset or is too complex. This PR solves this.
It also serves as a great comparison for a beginner to realize what Bevy is really like and how different it is from what they may expect Bevy to be. For example for someone coming from [LÖVE](https://love2d.org/), they will have something like this in their head when they think of drawing a rectangle:
```lua
function love.draw()
    love.graphics.setColor(0.25, 0.25, 0.75);
    love.graphics.rectangle("fill", 0, 0, 50, 50);
end
```
This, of course, differs quite a lot from what you do in Bevy. I imagine there will be people that just want to see something as simple as this in comparison to have a better understanding for the amount of differences.

## Solution

Add a dead simple example drawing a blue 50x50 rectangle in the center with no more and no less than needed.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
> If someone with more experience in Bevy could chime in, is it possible to draw anything other than a rectangle? Such as a circle, rounded rectangle, or triangle? With those this example could be graduated to "shapes" (mirroring the 3d_scene example) and would be very instructional for those who aren't good at drawing sprites :)

Currently, there are no other shapes supported out of the box :( `bevy_prototype_lyon` is the community's stopgap for now. I agree with this being a solid goal once that's in place.

**Comment by mockersf** [comment]
The new renderer has been merged, with a completely revamped `SpriteBundle`:
https://github.com/bevyengine/bevy/blob/49a4009d5cc06410b77707e56f5b43d11dfdf63a/crates/bevy_sprite/src/bundle.rs#L13-L23

Could you update this PR for the new bundle?

**Comment by r00ster91** [comment]
@mockersf I updated it hopefully correctly.

File: examples/2d/rect.rs [hunk 1]
```diff
@@ -10,8 +10,13 @@ fn main() {
 fn setup(mut commands: Commands, mut materials: ResMut<Assets<ColorMaterial>>) {
     commands.spawn_bundle(OrthographicCameraBundle::new_2d());
     commands.spawn_bundle(SpriteBundle {
-        material: materials.add(Color::rgb(0.25, 0.25, 0.75).into()),
-        sprite: Sprite::new(Vec2::new(50., 50.)),
-        ..Default::default()
+        transform: Transform {
+            scale: Vec3::new(50.0, 50.0, 0.0),
+            ..Default::default()
+        },
```
Comment (mockersf) [inline examples/2d/rect.rs:16]: why do you set the scale here? is it to replace the `Sprite::new(Vec2::new(50., 50.))`?

Then you could do like https://github.com/bevyengine/bevy/blob/7356f1586d74039f840bcfcf24af3e21c23e3c18/examples/2d/contributors.rs#L100

Comment (r00ster91) [reply examples/2d/rect.rs]: Ah, okay. That's what someone else suggested as well but neither of us were sure which one is the right way.
I tried something else now. Is it fine?

**Comment by mockersf** [comment]
looking good, thanks!

**Pull request closed without merging**